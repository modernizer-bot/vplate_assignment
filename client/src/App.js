import React from 'react';

import Modal from './components/hocs/Modal.jsx';
import Home from './components/pages/Home';

export default function App() {
  return (
    <Modal>
      <Home />
    </Modal>
  );
}
