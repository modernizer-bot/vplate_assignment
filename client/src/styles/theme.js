const color = {
  lightGray: '#eff3f4',
  darkGray: '8E99A1',
  twitter: '#31a4f1',
};

const size = {
  profileImageSize: '48px',
  deviceSize: '650px',
  headerSize: '130px',
};

const font = {
  small: '15px',
  medium: '20px',
  big: '30px',
};

const theme = {
  size,
  color,
  font,
};

export default theme;
