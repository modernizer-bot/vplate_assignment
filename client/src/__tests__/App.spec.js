import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import App from '../App';
import createStore from '../store';
import theme from '../styles/theme';

describe('App component test', () => {
  it('renders correctly', () => {
    // For test React.Portal
    ReactDOM.createPortal = (node) => node;

    const store = createStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <App />
          </ThemeProvider>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
