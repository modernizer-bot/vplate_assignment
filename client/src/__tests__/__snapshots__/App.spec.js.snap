// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test renders correctly 1`] = `
<div
  className="sc-bdvvtL"
>
  <div
    className="sc-gsDKAQ fsjGqn"
    onClick={[Function]}
  >
    <div
      className="sc-dkPtRN fDSPkq"
    />
    <div
      className="sc-hKwDye laLKBu"
    >
      <div
        className="sc-eCImPb SquWf"
      >
        <p
          className="sc-jRQBWg iWMFnb"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-jJoQJp eSIyXv"
  >
    <div
      className="sc-hiCibw kKsheI"
    >
      <div
        className="sc-kLwhqv eXkmeU"
      >
        <div
          className="sc-gKclnd bxOyCw"
        >
          <div
            className="sc-iCfMLu fdIwrv"
          >
            <svg
              color="#31a4f1"
              fill="currentColor"
              height={40}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "#31a4f1",
                }
              }
              viewBox="0 0 1024 1024"
              width={40}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0 0 75-94 336.64 336.64 0 0 1-108.2 41.2A170.1 170.1 0 0 0 672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 0 0-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 0 1-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 0 1-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
              />
            </svg>
          </div>
        </div>
        <div
          className="sc-ikJyIC htgCJA"
        >
          Assignment Project
        </div>
      </div>
      <div
        className="sc-furwcr iGJkXb"
      >
        <div
          className="sc-kDTinF bhskWv"
        >
          <div
            className="sc-pVTFL hMFaSS"
          >
            <div
              className="sc-gKclnd bxOyCw"
            >
              <div
                className="sc-iCfMLu fdIwrv"
              >
                <svg
                  color="black"
                  fill="currentColor"
                  height={24}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": "black",
                    }
                  }
                  viewBox="0 0 1024 1024"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            className="sc-jrQzAO lcFYhz"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Search Keyword"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="sc-gWXbKe eXqVCI"
    >
      <div
        className="sc-ezbkAF kbmPlw"
      >
        <p
          className="sc-bYoBSM fmYlnJ"
        >
          Tweet List is empty :(
        </p>
      </div>
    </div>
  </div>
</div>
`;
